src/hotspot/share/gc/shenandoah/shenandoahFreeSet.cpp

```diff
@@ -246,13 +246,45 @@ HeapWord* ShenandoahFreeSet::allocate_single(ShenandoahAllocRequest& req, bool&
   return nullptr;
 }
 
+size_t get_usable_free_words(size_t free) {
+  // e.g. card_size is 512, card_shift is 9, min_fill_size() is 8
+  //      free is 514
+  //      usable_free is 512, which is decreased to 0
+  size_t usable_free = (free / CardTable::card_size()) << CardTable::card_shift();
+  if ((free != usable_free) && (free - usable_free < ShenandoahHeap::min_fill_size() * HeapWordSize)) {
+    // We'll have to add another card's memory to the padding
+    if (usable_free >= CardTable::card_size()) {
+      usable_free -= CardTable::card_size();
+    } else {
+      assert(usable_free == 0, "usable_free is a multiple of card_size and card_size > min_fill_size");
+    }
+  }
+
+  return usable_free / HeapWordSize;;
+}
+
+HeapWord* ShenandoahFreeSet::allocated_aligned_plab(size_t size, ShenandoahAllocRequest& req, ShenandoahHeapRegion* r) {
```

Comment:
Did you really want the method to be called `allocated_...` or `allocate_...`.

I'd think the imperative verb form works better here, and I assume the extra `d` was an unintended typo?